src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -885,6 +889,176 @@ MavlinkReceiver::handle_message_set_position_target_local_ned(mavlink_message_t
 	}
 }
 
+void
+MavlinkReceiver::handle_message_set_position_target_global_int(mavlink_message_t *msg)
+{
+	mavlink_set_position_target_global_int_t set_position_target_global_int;
+	mavlink_msg_set_position_target_global_int_decode(msg, &set_position_target_global_int);
+
+	const bool values_finite =
+		PX4_ISFINITE(set_position_target_global_int.alt) &&
+		PX4_ISFINITE(set_position_target_global_int.vx) &&
+		PX4_ISFINITE(set_position_target_global_int.vy) &&
+		PX4_ISFINITE(set_position_target_global_int.vz) &&
+		PX4_ISFINITE(set_position_target_global_int.afx) &&
+		PX4_ISFINITE(set_position_target_global_int.afy) &&
+		PX4_ISFINITE(set_position_target_global_int.afz) &&
+		PX4_ISFINITE(set_position_target_global_int.yaw);
+
+	/* Only accept messages which are intended for this system */
+	if ((mavlink_system.sysid == set_position_target_global_int.target_system ||
+	     set_position_target_global_int.target_system == 0) &&
+	    (mavlink_system.compid == set_position_target_global_int.target_component ||
+	     set_position_target_global_int.target_component == 0) &&
+	    values_finite) {
+
+		offboard_control_mode_s offboard_control_mode{};
+
+		/* convert mavlink type (local, NED) to uORB offboard control struct */
+		offboard_control_mode.ignore_position = (bool)(set_position_target_global_int.type_mask & 0x7);
+		offboard_control_mode.ignore_alt_hold = (bool)(set_position_target_global_int.type_mask & 0x4);
+		offboard_control_mode.ignore_velocity = (bool)(set_position_target_global_int.type_mask & 0x38);
+		offboard_control_mode.ignore_acceleration_force = (bool)(set_position_target_global_int.type_mask & 0x1C0);
+		offboard_control_mode.ignore_attitude = (bool)(set_position_target_global_int.type_mask & 0x400);
+		offboard_control_mode.ignore_bodyrate_x = (bool)(set_position_target_global_int.type_mask & 0x800);
+		offboard_control_mode.ignore_bodyrate_y = (bool)(set_position_target_global_int.type_mask & 0x800);
+		offboard_control_mode.ignore_bodyrate_z = (bool)(set_position_target_global_int.type_mask & 0x800);
+
+		/* yaw ignore flag mapps to ignore_attitude */
+		bool is_force_sp = (bool)(set_position_target_global_int.type_mask & (1 << 9));
+
+		bool is_takeoff_sp = (bool)(set_position_target_global_int.type_mask & 0x1000);
+		bool is_land_sp = (bool)(set_position_target_global_int.type_mask & 0x2000);
+		bool is_loiter_sp = (bool)(set_position_target_global_int.type_mask & 0x3000);
+		bool is_idle_sp = (bool)(set_position_target_global_int.type_mask & 0x4000);
```

Comment:
You can move these calculations to later where you actually require them. And then you can set the `position_sp_triplet.current.type` directly without having to use an intermediate `bool is_...`.

Reply:
Regarding those type_masks, I was following what was done for `SET_POSITION_TARGET_LOCAL_NED` but I couldn't find any documentation on how those typemasks were defined. @hamishwillee FYI